src/s390_crypto.c

```diff
@@ -756,3 +760,32 @@ int s390_get_functionlist(libica_func_list_element *pmech_list,
   }
   return 0;
 }
+
+#ifdef ICA_FIPS
+int s390_get_fips_indicator(libica_fips_indicator_element *indicator_list,
+						unsigned int *indicator_list_len)
+{
+	size_t i;
+
+	if (!indicator_list_len) {
+		return EINVAL;
+	}
+
+	if (!indicator_list) {
+		*indicator_list_len = sizeof(icaList) / sizeof(libica_fips_indicator_element);
+		return 0;
+	}
+
+	if (*indicator_list_len < (sizeof(icaList) / sizeof(libica_fips_indicator_element))) {
+		return ENOMEM;
```

Comment:
ENOMEM? I would use EINVAL here. ENOMEM is for allocating memory only.

Reply:
ok, same discussion as above ... if I change it above, then also here.